apps/src/applab/AppLabView.jsx

```diff
@@ -173,8 +164,16 @@ var AppLabView = React.createClass({
     return (
       <ConnectedStudioAppWrapper>
         <div id="visualizationColumn">
-          {playSpaceHeader}
-          <ProtectedStatefulDiv contentFunction={this.props.generateVisualizationColumnHtml} />
+          {!this.props.isReadOnlyWorkspace && <PlaySpaceHeader
+              isEditingProject={this.props.isEditingProject}
+              screenIds={this.props.screenIds}
+              onViewDataButton={this.props.onViewDataButton}
+              onScreenCreate={this.props.onScreenCreate} />
+          }
+          <ApplabVisualizationColumn
+              isProjectLevel={this.props.isProjectLevel}
+              isSubmittable={this.props.isSubmittable}
+              isSubmitted={this.props.isSubmitted}/>
```

Comment:
Agreed that PlaySpaceHeader probably belongs inside of ApplabVisualizationColumn.

Because you can connect anything at any level (as long as the very top level is wrapped) I've been connecting lower/smaller components, and finding that often the structural/larger components don't need to be connected.  I find this is a satisfying separation of concerns, where my layout glue code no longer has to be my state glue code too.

If AppLabView already needs all three props that ConnectedStudioAppWrapper needs, maybe it does make sense to just pass the props into StudioAppWrapper.  But if CSAW needs something ALV doesn't, why give ALV more knowledge of the world than it needs?

The redux guidance toward container/presentational components seems to promote building components in pairs so that you keep the connection logic separate from the presentational part at each layer - and I _think_ just by using the `react-redux` `connect()` helper everywhere we are following that pattern.  Something like this: